rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/CatalogResource.java

```diff
@@ -331,6 +332,10 @@ public void deleteEntity(String symbolicName, String version) throws Exception {
             throw WebResourceUtils.forbidden("User '%s' is not authorized to modify catalog",
                 Entitlements.getEntitlementContext().user());
         }
+
+        if (LATEST.equals(version)) {
```

Comment:
Good catch. Addressed in https://github.com/apache/brooklyn-server/pull/635/commits/361a0fd47377d012d8b484dc925b4da3695887d3